Lib/test/support/testresult.py

```diff
@@ -0,0 +1,201 @@
+'''Test runner and result class for the regression test suite.
+
+'''
+
+import functools
+import io
+import sys
+import time
+import traceback
+import unittest
+
+import xml.etree.ElementTree as ET
+
+from datetime import datetime
+
+class RegressionTestResult(unittest.TextTestResult):
+    separator1 = '=' * 70 + '\n'
+    separator2 = '-' * 70 + '\n'
+
+    def __init__(self, stream, descriptions, verbosity):
+        super().__init__(stream=stream, descriptions=descriptions, verbosity=0)
+        self.buffer = True
+        self.__suite = ET.Element('testsuite')
+        self.__suite.set('start', datetime.utcnow().isoformat(' '))
+
+        self.__e = None
+        self.__start_time = None
+        self.__results = []
+        self.__verbose = bool(verbosity)
```

Comment:
Inheritance - some of these may (either now or in future) conflict with those in the base class. Particularly because it's derived from `TextTestResult`, which overrides some of the `TestResult` functionality that kind of matches what this brings back.